compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -254,6 +254,8 @@ private static boolean containsTypeParam(BType type, HashSet<BType> resolvedType
                 return containsTypeParam(errorType.detailType, resolvedTypes);
             case TypeTags.TYPEDESC:
                 return containsTypeParam(((BTypedescType) type).constraint, resolvedTypes);
+            case TypeTags.TYPEREFDESC:
+                return containsTypeParam(Types.getReferredType(type), resolvedTypes);
```

Comment:
We can do `type = Types.getReferredType(type)` at the top and remove this case. Please update similar cases if any

Reply:
Updated b94a3ac3618939988d528db9ce32b01cc3d02fa9